### ISSUE TITLE
World Poker Tour from Stern still the lights don't work after vpinmame 720 (released in Jan 3)

### ISSUE THREAD

**Issue by jpsalas59** [vpinball/pinmame#350]
Something changed between 720 and 797. On some computers any revision after 720 the lights in World Poker Tour stop working after a second. The game works, but there are no lights working,

I see there was an issue before, #271 , and some roms were fixed, but still there are problems with this World Poker Tour.

Not all computers are affected. I have 2 computers, one with the Nvidia 3080 card works fine, but the one with an Intel Iris doesn't work. Both computers are using Windows 11 with the same updates, and the same VPX and vpinmame.

**Comment by jpsalas59** [comment]
I made a new table using this rom, and I didn't use any of the code using arrays for the lights, so this table can be used for testing;

https://www.vpforums.org/index.php?showtopic=54314

**Comment by toxieainc** [comment]
@vbousquet may know what happened/broke

**Comment by vbousquet** [comment]
I was not able to reproduce on my hand so far. @jpsalas59 do you have the issue on x86 build as well as x64 or only x64 ?
It may be related to https://github.com/vpinball/pinmame/issues/311 (some timing change ? uninitialized mem ?)

**Comment by jpsalas59** [comment]
I don't think it is related to #311 because that is an out of sync problem when using the 64 bits vpinmame, the lack of that at91Jit emulation which is only 32 bits.

With my desktop computer works all the time, 32 bits, 64 bits and GL. But with my portable I have the error all the time, and this is the lights stop working just when the Stern Leds turn on., this is just a few seconds after the game has started. And this happen in 32 bits and 64 bits, normal DX and GL, so the problem is not VPX8, but vpinmame, as it works perfectly with the vpinmame from  VPinMAME-sc-3.6-720 and older. I may be a timing problem. There is no visible error, just the lights stops updating. I made the table World Joker Tour mostly to test this and do not use the old fading routines, and I used a very simple script, using mostly the core.vbs routines.

**Comment by GHOSTfaceKILLr1974** [comment]
> @vbousquet may know what happened/broke

It might be related so I'm commenting the issue here #350. I used to have problems with flashers/lamps not lighting up at all with "World Poker Tour" table. I updated to VPX 10.8.0-2027 x64, VPinMAME 3.6.0-1150 x64 and now the flashers/lamps always stay on with "World Poker Tour", "Wheel of Fortune" and "The Walking Dead" tables.

**Comment by apinbalwiz** [comment]
I downgraded from VPX 10.8 2028 x64, VPinMAME 3.6 1152 x64 to try 10.8 2027 x64, VPinMAME 3.6 1150 x64 and unfortunately World Poker Tour still doesn't work (I haven't had a problem with Wheel of Fortune or The Walking Dead with any build).

**Comment by vbousquet** [comment]
@jpsalas59 I had a look and found an issue that could eventually be the cause of this bug. Could you give a try to the latest VPinMame and see if the bug is still there (as I can not be sure on my side, since I never was able to reproduce) ?

**Comment by jpsalas59** [comment]
Thanks! I'll check it out and I'll send it also to a friend who has this problem.

**Comment by jpsalas59** [comment]
In my portable computer the WPT ROM started working when I updated the C++ runtimes, using a few weeks older vpinmame. But my friend Leo tried that and it didn't work. He tried now with this latest vpinmame, and it works just the same as before, the lights blink for a few seconds and then they stop updating, This is the same error as other people were having. Now I do not have any computer with this error anymore.

**Comment by GHOSTfaceKILLr1974** [comment]
I noticed a vpimame update involving WPT. I tried it and still the lights won't flash.
Using freneticamnesic WPT table.

**Comment by apinbalwiz** [comment]
I tried it using vpimame 1185 and vpinball RC6, as well as updating to the latest C++ runtimes and the WPT lighting issue remains.....

**Comment by vbousquet** [comment]
@apinbalwiz @GHOSTfaceKILLr1974 I just pushed a new update that could hopefully also fix this issue. Could you give it a try (I never had the issue on my end, so it's difficult for me to say if things are good or not) ?

**Comment by jpsalas59** [comment]
I just tested it and it works very well on my two computers. 
Just a note: on my desktop computer all of a sudden it stopped working, and after trying to fix it I found out I could do a Windows reinstall, since it costs nothing to me do it. And the rom started working again. Same Windows 11 version, same VPX and same vpinmame (both the latest, for about a week ago). I did the same reinstall with the portable and the WPT rom started working again. I didn't reinstall any VP related apps, just run the pinmame setup.exe to register vpinmame.dll. I don't know anymore what could be wrong, I thought it could be a cache problem, but that January revision 720, still worked fine. Anyway a Windows reinstall does the trick :) I told my friend in Argentina with the same problem and he did a Windows 10 reinstall and that helped too.

**Comment by toxieainc** [comment]
Thats kinda spooky though  ;)

**Comment by GHOSTfaceKILLr1974** [comment]
> @apinbalwiz @GHOSTfaceKILLr1974 I just pushed a new update that could hopefully also fix this issue. Could you give it a try (I never had the issue on my end, so it's difficult for me to say if things are good or not) ?

Looks like it's kind of working (using freneticamnesic WPT table and tried bigus modded version). It still rarely flashes when the ball hits the slingshot but now flashes when in multiball, hitting jackpots and does its flashing display after the game ends.
Note: I'm not completely familiar with this game but Definitely see a noticeable Improvement.

**Comment by jpsalas59** [comment]
Today I tried the latest VPX8 10.8.1, and somehow it provoked the error. After some registry editing and deleting some cache files I found out that the problem is the vpinball.ini in the %appdata%\VPinballX
Something is written to the vpinball.ini that makes this rom stop updating the lights. I have spared the bad vpinball.ini so it is very easy to reproduce. I send you a zip file with the bad and the good ini files. I have searched for the changes, but I couldn't find where the problem is.
The strange thing is that the latest VPX 7.4 is also affected by the bug, and that VPX should not use the ini file (?!)

[vpinball_ini.zip](https://github.com/user-attachments/files/18467365/vpinball_ini.zip)

**Comment by Soundequ** [comment]
@jpsalas59
I'm a total dope without fancy tools, but using your working .ini.good and my own bad .ini I did some section for section and then line for line brute force-y (copy/overwrite) checking and narrowed it down (at least for me?) to a B2S problem.

Controller preferences section (line 354 on your file)
ForceDisableB2S = 1 -- insert lights continue.
...but turn that off (0) and they die after a couple seconds as reported.

Much as I like this table, obviously I don't want to disable B2S for every other table.
Thanks to you (all) for doing 99.99(repeating)% of the work!

**Comment by jpsalas59** [comment]
Thanks Soundequ! That's was the difference! If I enable the B2s then the lights stop, and if I disable the B2s, the lights work again :) At last after many months, the error is easy to reproduce :)

**Comment by GHOSTfaceKILLr1974** [comment]
> Thanks Soundequ! That's was the difference! If I enable the B2s then the lights stop, and if I disable the B2s, the lights work again :) At last after many months, the error is easy to reproduce :)

So the issue is not with vpinmame but with vpx?
Also do I fix this in the vpinball.ini file or go to "Key, Nudge and DOF" menu and do I check or uncheck "Disable B2S"?
I'm currently at work so I can't try.

**Comment by Soundequ** [comment]
I just (copy/paste) manually edited the .ini because I was searching for the problem. Now that the problem has been isolated/identified, you can do it either way.

**Comment by jpsalas59** [comment]
> (I only have issues with the flashers on the slingshots)
GHOSTfaceKILLr1974: that's a problem with the table you are using. It uses my old fading lights routine, and it is using the light numbers 122, 123, 132 and 133 for the flashers on the slingshots. On the old vpinmame those numbers were not used, but now the rom controls those numbers, that's why the flashers do not always work as intended, as those numbers now belongs to lights, and the rom may turn them on or off as it pleases. The code for the flashers need to be updated in that table, The same goes for the Bigus mods. So, it is not a vpinmame issue, it is a table issue.

**Comment by apinbalwiz** [comment]
> [@apinbalwiz](https://github.com/apinbalwiz) [@GHOSTfaceKILLr1974](https://github.com/GHOSTfaceKILLr1974) I just pushed a new update that could hopefully also fix this issue. Could you give it a try (I never had the issue on my end, so it's difficult for me to say if things are good or not) ?

Vbousquet I just saw your message, I'm sorry that I missed it and test the new update....

I'm glad to see that a work around was found.  Is there a way to disable b2s for individual tables and not globally? The vpinball.ini is in the registry is it not?
